core/src/main/java/org/apache/iceberg/hadoop/HadoopTableOperations.java

```diff
@@ -152,6 +154,19 @@ public void commit(TableMetadata base, TableMetadata metadata) {
     this.shouldRefresh = true;
   }
 
+  /**
+   * Deletes the file from the file system. Any RuntimeIOException will be swallowed and logged.
+   *
+   * @param tempMetadataFile the file to be deleted.
+   */
+  private void deleteFileNoThrow(final Path tempMetadataFile) {
```

Comment:
Instead of catching a runtime IO exception and logging it, I think this should catch any runtime exception and add it as a suppressed exception to the CommitFailedExeception.

Reply:
Thanks for the review. Updated PR to suppress. Also refactored the rename code to a separate method because I'll need to change the throw line of `CommitFailedException` anyways.